### ISSUE TITLE
Prognostic run report: Fails if first run is not the longest run

### ISSUE THREAD

**Issue by brianhenn** [ai2cm/fv3net#1758]
TLDR: The report will fail with an unhelpful error if the first run isn't (at least tied for) the longest run, which is a gotcha. Some more robust handling of missing variables in `RunDiagnostics.get_variables` might be helpful.

The following prognostic run report configuration in `rundirs.json` will fail:
```
[
  {
    "url": "some_path",
    "name": "a-10-day-run"
  },
  {
    "url": "some_other_path",
    "name": "a-40-day-run"
  },
]
```
with the following:
```
"Traceback (most recent call last):
  File "/opt/conda/envs/fv3net/bin/prognostic_run_diags", line 33, in <module>
    sys.exit(load_entry_point('fv3net-diagnostics', 'console_scripts', 'prognostic_run_diags')())
  File "/home/jovyan/fv3net/workflows/diagnostics/fv3net/diagnostics/prognostic_run/cli.py", line 38, in main
    args.func(dissoc(args, "func"))
  File "/home/jovyan/fv3net/workflows/diagnostics/fv3net/diagnostics/prognostic_run/views/static_report.py", line 640, in main_json
    make_report(computed_diagnostics, args.output)
  File "/home/jovyan/fv3net/workflows/diagnostics/fv3net/diagnostics/prognostic_run/views/static_report.py", line 564, in make_report
    "maps.html": render_maps(metadata, diagnostics, metrics),
  File "/home/jovyan/fv3net/workflows/diagnostics/fv3net/diagnostics/prognostic_run/views/static_report.py", line 448, in render_maps
    time_mean_cubed_sphere_maps(diagnostics, metrics),
  File "/home/jovyan/fv3net/workflows/diagnostics/fv3net/diagnostics/prognostic_run/views/static_report.py", line 243, in time_mean_cubed_sphere_maps
    return plot_cubed_sphere_map(
  File "/home/jovyan/fv3net/workflows/diagnostics/fv3net/diagnostics/prognostic_run/views/matplotlib.py", line 136, in plot_cubed_sphere_map
    ds = run_diags.get_variables(run, COORD_VARS + [varname])
  File "/home/jovyan/fv3net/workflows/diagnostics/fv3net/diagnostics/prognostic_run/computed_diagnostics.py", line 146, in get_variables
    return xr.merge(variables)
  File "/opt/conda/envs/fv3net/lib/python3.8/site-packages/xarray/core/merge.py", line 900, in merge
    merge_result = merge_core(
  File "/opt/conda/envs/fv3net/lib/python3.8/site-packages/xarray/core/merge.py", line 635, in merge_core
    variables, out_indexes = merge_collected(
  File "/opt/conda/envs/fv3net/lib/python3.8/site-packages/xarray/core/merge.py", line 240, in merge_collected
    merged_vars[name] = unique_variable(name, variables, compat)
  File "/opt/conda/envs/fv3net/lib/python3.8/site-packages/xarray/core/merge.py", line 149, in unique_variable
    raise MergeError(
xarray.core.merge.MergeError: conflicting values for variable 'final_time' on objects to be combined. You can skip this check by specifying compat='override'."
```

What is happening is that 1) the 'verification' entry is computed off of the longest run (40 days), 2) some variables are plotted as maps that are not in the 'verification' dataset (eg, `tmpsfc_time_mean_value`, see #1158 ), 3) in retrieving the variable itself plus the grid variables to make the verification map, the `RunDiagnostics.get_variables` function will try to load the missing variable but won't find it in the verification dataset, so will try to load it in json-specified sequence from the other datasets, leading to it loading it from the first (10-day) dataset, 4) the grid vars are in the verification dataset and so are from the 40-day run, and finally 5) `get_variables` will merge the variable of interest and the grid variables, their `final_time` coordinate variable will be different, and the merge will fail. 

Fixes -- a quick one is to drop `final_time` in `RunDiagnostics.get_variables` though that seems like a bandaid. Maybe `RunDiagnostics.get_variables` should go looking for missing variables in the other run datasets sequentially by run length, since this is how the verification is done?

**Comment by oliverwm1** [comment]
I just ran into this, thanks for diagnosing. Maybe we can just drop the singleton `time` dimension (could keep is an attr) instead of renaming it from `time` -> `final_time`. Probably have to do something similar for `initial_time`?

https://github.com/ai2cm/fv3net/blob/1f9cbccd66460598407abd5f8787a20980be4700/workflows/diagnostics/fv3net/diagnostics/prognostic_run/compute.py#L544-L549

**Comment by oliverwm1** [comment]
I took a look at one of the netCDFs produced by the prog run diagnostics. Every data variable includes `initial_time` and `final_time` scalar coordinates, even the grid variables like `lat` etc., which feels a little wrong. Perhaps we can just assign these as global attributes? Slight downside being we'd need to encode them as string.

**Comment by brianhenn** [comment]
Making the final time and initial time dataset-level attributes sounds good. Do we even use them for anything?

**Comment by oliverwm1** [comment]
> Making the final time and initial time dataset-level attributes sounds good. Do we even use them for anything?

No, each diagnostic which is a reduction over time has its own attribute that describes the reduction period.

With the PR I linked, we'll just no longer have these coords or save them as attrs.
